contracts/lens/lib.rs

```diff
@@ -62,6 +67,7 @@ pub mod contract {
     #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
     pub struct PoolUnderlyingPrice {
         pool: AccountId,
+        /// underlying_price of the pool
```

Comment:
Comments like this aren't all that useful to readers since it doesn't provide any extra context.

You could, for example, provide information like where the price is used, units, considerations when doing calculations with it, etc.